Update SSO information (#116)
doitintl/help: website/docs/user-management/authentication.mdx

---
sidebar_position: 5
---

# Authentication

To configure authentication for your organization, select the gear icon from the top navigation bar of the _Cloud Management Platform_ (CMP), then select **Authentication**:

<picture
  img={require('../../assets/png/cmp-settings-auth.png')}
  alt="A screenshot of the Authentication screen"
  frame="drop-shadow"
/>

From here, you can configure two types of authentication flow:

> - **Single Sign-On**: Use a custom _Single Sign-On_ (SSO) application, also known as _Identity Provider_ (IdP) initiated SSO

Reviewer: The sso page isn't related to whether the user will choose IDP or SP flow. It's just to configure their sso app integration

Author: Same comment as:

- https://github.com/doitintl/help/pull/116#discussion_r916750648

> - **Auth provider**: Restrict [user log in](../general/log-in.mdx) via Google Workspace or Microsoft Office 365, also known as _Service Provider_ (SP) initiated SSO

Reviewer: SP initiated isn't related to this page at all

Author: Ah okay, thanks. This is the one thing about your review that I don't think I understand. We can discuss on a call when you're available!

  alt="A screenshot of the OIDC form field with example information"
  frame="drop-shadow"
/>

After selecting **Configure**, the CMP will present the following form:

<picture
  img={require('../../assets/png/cmp-settings-auth-oidc.png')}
  alt="A screenshot of the SAML protocol configuration form"
  frame="drop-shadow"
/>

To configure the SSO for the CMP with the OIDC protocol, you must provide the following information about your OIDC application:

- **Client ID**: Your application's _Client ID_
- **Issuer URL**: Your application's _Issuer URL_ (aka the metadata _Discovery URL_)
- **Client secret**: Your application's _Client Secret_

The CMP will prompt you to confirm the operation when you select the **Save** button. If you select **Confirm**, the CMP will automatically enable SSO:

<picture
  img={require('../../assets/png/cmp-settings-auth-sso-oidc.png')}
  alt="A screenshot of the Authentication screen with SSO enabled and OIDC configured"
  frame="drop-shadow"
/>

:::info See also

- [Otka: Create OIDC app integrations using AIW](https://help.okta.com/oie/en-us/Content/Topics/Apps/Apps_App_Integration_Wizard_OIDC.htm)

> - [Google Cloud Platform: Signing in users with OIDC](https://cloud.google.com/identity-platform/docs/web/oidc)

Reviewer: Same as above, not relevant

- [Microsoft: OpenID Connect authentication with Azure Active Directory](https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/auth-oidc)

:::

### Change protocols

If you have configured both **SAML** and **OIDC**, you can change the protocol by selecting the corresponding radio button. The CMP will prompt you to confirm the change:

<picture
  img={require('../../assets/png/cmp-settings-auth-sso-change.png')}
  alt="A screenshot of the Auth provider screen"
  frame="drop-shadow"
/>

### User roles

You can configure user roles via your Identity Provider (IdP) or [directly from within the CMP](../user-management/manage-roles.mdx).

> If set, the CMP will assign roles to users based on the `doit_platform_role_id` attribute, as provided by your IdP. The value of this attribute should exactly match the name of [the desired CMP user role](../user-management/manage-roles.mdx).

Reviewer: the value of `doit_platform_role_id` should be the role id, not the name

Author: Oh wow, thanks. I was confused by the ticket describing this because we don't document the _Role ID_ in the docs and I had to go double check what you meant in the CMP. I found the button and have added documentation with a screenshot.

If your IdP does not provide a `doit_platform_role_id` attribute, the CMP will assign users the [default role](../user-management/manage-roles.mdx#setting-a-default-role) configured for your organization.

## Auth provider

By default, the CMP supports _Service Provider_ (SP) initiated Single Sign-On (SSO) for both Google Workspace and Microsoft Office 365 accounts.

> If you would like to restrict SSO to a specific provider, select **Auth provider** from the left-hand menu on the **Authentication** screen and choose the corresponding option:

Reviewer: That isn't true

Author: Same comment/issue as:

- https://github.com/doitintl/help/pull/116#discussion_r916750648 (i.e., let's discuss)

---
sidebar_position: 5
---

# Authentication

To configure authentication for your organization, select the gear icon from the top navigation bar of the _Cloud Management Platform_ (CMP), then select **Authentication**:

<picture
  img={require('../../assets/png/cmp-settings-auth.png')}
  alt="A screenshot of the Authentication screen"
  frame="drop-shadow"
/>

> From here, you can configure two types of authentication flow:

Author: Changed to:

> From here, you can configure two aspects of authentication:

> From here, you can configure two types of authentication flow:

Reviewer: perfect

- **Single Sign-On**: Use your own _Single Sign-On_ (SSO) application
- **Auth provider**: Allow [user log-ins](../general/log-in.mdx) via Google Workspace, Microsoft Office 365, or both

## Single Sign-On

The CMP supports the following _Identity Provider_ (IdP) protocols for connecting with your SSO application:

- _Security Assertion Markup Language_ (SAML)
- _OpenID Connect_ (OIDC)

:::info See also

- [Okta: What's the Difference Between OAuth, OpenID Connect, and SAML?](https://www.okta.com/uk/identity-101/whats-the-difference-between-oauth-openid-connect-and-saml/)

:::

You can configure the **SAML** or **OIDC** protocols from the **Single Sign-On** screen by selecting the corresponding **Configure** link.

:::note

You must configure **SAML** or **OIDC** before you can enable SSO with the **Enable SSO** radio button.

:::

### Configure SAML

> If your SSO application uses SAML, you must configure your chosen _Service Provider_ (SP) with the values displayed in the form field on the **Authentication** screen. For example:

Reviewer: Something doesn't make sense here. The CMP is the SP here so saying your chosen SP is weird. 
```suggestion
If you choose to use SAML you should use the values displayed in the form fields in your SSO application **Authentication** screen. For example:
```

<picture
  img={require('../../assets/png/cmp-settings-auth-saml-values.png')}
  alt="A screenshot of the SAML form field with example information"
  frame="drop-shadow"
/>

After selecting **Configure**, the CMP will present the following form:

<picture
  img={require('../../assets/png/cmp-settings-auth-saml.png')}
  alt="A screenshot of the SAML protocol configuration form"
  frame="drop-shadow"
/>

> To configure the SAML protocol, enter the following information, provided to you by your SP:

Reviewer: Should be IDP, not SP

- **Entity ID**: Your application's _Entity ID_ (aka _Audience URI_)
- **SSO URL**: Your application's _SSO URL_ (aka the _Destination URL_)
- **Certificate**: Your application's X.509 signing certificate

The CMP will prompt you to confirm the operation when you select the **Save** button. If you select **Confirm**, the CMP will automatically enable SSO:

<picture
  img={require('../../assets/png/cmp-settings-auth-sso-saml.png')}
  alt="A screenshot of the Authentication screen with SSO enabled and SAML configured"
  frame="drop-shadow"
/>

:::info See also

- [Okta: Create SAML app integrations using AIW](https://help.okta.com/oie/en-us/Content/Topics/Apps/Apps_App_Integration_Wizard_SAML.htm)
- [Google Workspace: Set up your own custom SAML application](https://support.google.com/a/answer/6087519?hl=en)
- [Microsoft: SAML authentication with Azure Active Directory](https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/auth-saml)

:::

### Configure OIDC

> If your SSO application uses OIDC, you must configure your chosen _Service Provider_ (SP) with the values displayed in the form field on the **Authentication** screen. For example:

Reviewer: Same as the comment above

<picture
  img={require('../../assets/png/cmp-settings-auth-oidc-values.png')}
  alt="A screenshot of the OIDC form field with example information"
  frame="drop-shadow"
/>

After selecting **Configure**, the CMP will present the following form:

<picture
  img={require('../../assets/png/cmp-settings-auth-oidc.png')}
  alt="A screenshot of the SAML protocol configuration form"
  frame="drop-shadow"
/>

> To configure the SAML protocol, enter the following information, provided to you by your SP:

Reviewer: IDP, not SP

- **Issuer URL**: Your application's _Issuer URL_ (aka the metadata _Discovery URL_)
- **Client secret**: Your application's _Client Secret_

The CMP will prompt you to confirm the operation when you select the **Save** button. If you select **Confirm**, the CMP will automatically enable SSO:

<picture
  img={require('../../assets/png/cmp-settings-auth-sso-oidc.png')}
  alt="A screenshot of the Authentication screen with SSO enabled and OIDC configured"
  frame="drop-shadow"
/>

:::info See also

- [Okta: Create OIDC app integrations using AIW](https://help.okta.com/oie/en-us/Content/Topics/Apps/Apps_App_Integration_Wizard_OIDC.htm)
- [Microsoft: OpenID Connect authentication with Azure Active Directory](https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/auth-oidc)

:::

### Change protocols

If you have configured both **SAML** and **OIDC**, you can change the protocol by selecting the corresponding radio button. The CMP will prompt you to confirm the change:

<picture
  img={require('../../assets/png/cmp-settings-auth-sso-change.png')}
  alt="A screenshot of the Auth provider screen"
  frame="drop-shadow"
/>

### User roles

> You can configure user roles via your Identity Provider (IdP) or [directly from within the CMP](../user-management/manage-roles.mdx).

Reviewer: Let's drop this `or [directly from within the CMP]` it is the situation right now but in a week or so we gonna disable it.
